### ISSUE TITLE
Long sentence take too much memory

### ISSUE THREAD

**Issue by 333caowei** [tensorflow/tensorflow#6298]
I use one layer LSTM to train my data.
I use Dynamic_rnn, rnn_size=128, num_layers=1, seq_max_length=2500, batch_size=10, embedding_size=128, softmax_size=1600.

My code like this:

> x_vec = tf.nn.embedding_lookup(embedding_matrix_variable, self.x)
> lstm_fw_cell = rnn_cell.LSTMCell(num_units = hidden_unit, input_size = hidden_unit)
> lstm_fw_cell = rnn_cell.DropoutWrapper(lstm_fw_cell, output_keep_prob=self.dropout_keep_prob, input_keep_prob=self.dropout_keep_prob)
> outputs, _ = rnn.dynamic_rnn(lstm_fw_cell, x, dtype=tf.float32, sequence_length=real_length, swap_memory=False)
> 

I specify the GPU like this:
<img width="281" alt="screenshot" src="https://cloud.githubusercontent.com/assets/4569055/21170698/3bbf5de8-c20a-11e6-9b66-647830bdd371.png">

Command "nvidia-smi" shows as:
<img width="511" alt="screenshot" src="https://cloud.githubusercontent.com/assets/4569055/21170316/7738d3b6-c207-11e6-860a-cdc5ea166f40.png">

After lunching the program, it always shows :

> "I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator:...............":

details show as below:

> Step: 0       (epoch: 0.00000)       time:28.85937s
> Minibatch loss:          7.43977     Minibatch accuracy:  0.00000
> Test loss:         nan     Test accuracy: 0.00000
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 0 get requests, put_count=9012 evicted_count=9000 eviction_rate=0.998668 and unsatisfied allocation rate=0
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 9559 get requests, put_count=15572 evicted_count=6000 eviction_rate=0.385307 and unsatisfied allocation rate=0
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 25809 get requests, put_count=41821 evicted_count=16000 eviction_rate=0.382583 and unsatisfied allocation rate=3.87462e-05
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 42059 get requests, put_count=68071 evicted_count=26000 eviction_rate=0.381954 and unsatisfied allocation rate=2.37761e-05
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 58309 get requests, put_count=94321 evicted_count=36000 eviction_rate=0.381675 and unsatisfied allocation rate=1.715e-05
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 0 get requests, put_count=8014 evicted_count=8000 eviction_rate=0.998253 and unsatisfied allocation rate=0
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 7817 get requests, put_count=25828 evicted_count=18000 eviction_rate=0.696918 and unsatisfied allocation rate=0.000383779
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 24067 get requests, put_count=52078 evicted_count=28000 eviction_rate=0.537655 and unsatisfied allocation rate=0.000124652
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 40317 get requests, put_count=78328 evicted_count=38000 eviction_rate=0.485139 and unsatisfied allocation rate=7.44103e-05
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 56567 get requests, put_count=104578 evicted_count=48000 eviction_rate=0.458988 and unsatisfied allocation rate=5.30345e-05
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 0 get requests, put_count=7016 evicted_count=7000 eviction_rate=0.997719 and unsatisfied allocation rate=0
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 6070 get requests, put_count=23084 evicted_count=17000 eviction_rate=0.736441 and unsatisfied allocation rate=0.000329489
> I tensorflow/core/common_runtime/gpu/pool_allocator.cc:244] PoolAllocator: After 22320 get requests, put_count=49333 evicted_count=27000 eviction_rate=0.547301 and unsatisfied allocation rate=0.000134409
>

**Comment by 333caowei** [comment]
Finally I find the answer, If I use "config.gpu_options.allow_growth = True", it will shows the real memory usage of GPU
